dev/n.ts

```diff
@@ -0,0 +1,175 @@
+import node from "./node";
+
+const versionArg: Fig.Arg = {
+  name: "version",
+  suggestions: [
+    {
+      name: ["latest", "current"],
+      description: "Newest official release",
+    },
+    {
+      name: ["lts"],
+      description: "Newest Long Term Support official release",
+    },
+    {
+      name: ["auto"],
+      description:
+        "Read version from file: .n-node-version, .node-version, .nvmrc, or package.json",
+    },
+    {
+      name: ["engine"],
+      description: "Read version from package.json",
+    },
+  ],
+  generators: [
+    {
+      script: "n lsr --all",
+      postProcess: function (out) {
+        const versions = out
+          .split("\n")
+          .slice(1)
+          .map((i) =>
+            i.split(".").map((_, i, a) => a.slice(0, i + 1).join("."))
+          )
+          .flat();
+        return [...new Set(versions)].map((version) => {
+          return {
+            name: [version, `v${version}`],
+            description: `Node.js ${version}`,
+          };
+        });
```

Comment:
perf: using a set from the beginning and avoiding redundant operations on strings is a good boost.
```suggestion
        const set = new Set()
        const versions = out
          .split("\n")
          .slice(1)
        for (const version of versions) {
          set.add(version); // 16.1.2
          const splitted = version.split(".")
          set.add(splitted[0] + "." + splitted[1]); // 16.1
          set.add(splitted[0]); // 16
        }
        return Array.from(set).map((version) => {
          return {
            name: [version, `v${version}`],
            description: `Node.js ${version}`,
          };
        });
```

Reply:
Very good idea to use a set from the start
 I had originally not implemented the splitting at all, so didn't use a Set... I had just quickly hacked that in afterwards, which wasn't the wisest idea.